### PR TITLE
Replace Azure Pipelines with GitHub Actions

### PR DESCRIPTION
**Pull request by ras0219-msft**
See run: https://github.com/ras0219-msft/vcpkg-tool/actions/runs/4824887104/jobs/8595194913

This PR introduces a GitHub Actions workflow to replace our Azure DevOps PR validation pipeline. Along the way, this PR also introduces `CMakePresets.json`, enabling consistent builds across dev machines and CI.

- [x] build failure https://github.com/ras0219-msft/vcpkg-tool/actions/runs/4835288003/jobs/8617439065
- [x] formatting failure, and ability to download the resulting diff https://github.com/ras0219-msft/vcpkg-tool/actions/runs/4835416570
- [x] vcpkg-test failure https://github.com/ras0219-msft/vcpkg-tool/actions/runs/4835296006
- [x] vcpkg-artifacts test failure https://github.com/ras0219-msft/vcpkg-tool/actions/runs/4835305275
- [x] e2e test failure https://github.com/ras0219-msft/vcpkg-tool/actions/runs/4835299620

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+          name: 'format.diff'
+          path: ${{ env.DiffFile }}
+      - name: Run vcpkg unit tests
+        run: ctest --preset ${{ matrix.preset }}
+        env:
+          VCPKG_ROOT: UNIT_TESTS_SHOULD_NOT_USE_VCPKG_ROOT
+      - name: Run vcpkg-artifacts unit tests
+        run: |
+          cd out/build/${{ matrix.preset }}/vcpkg-artifacts
+          node node_modules/mocha/bin/mocha --config mocha-config.yaml
```
Comment (BillyONeal) [inline .github/workflows/build.yaml:84]: No change requested for now: should we hook mocha up as a test so you don't need the second step here?

**Comment by BillyONeal** [comment]
Should this delete CMakeSettings.json?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+      - name: Configure and Build
+        if: matrix.os != 'windows-2022'
+        run: |
+          cmake --preset ${{ matrix.preset }}
+          cmake --build --preset ${{ matrix.preset }} -- -k0
+      - name: Configure and Build (Windows)
+        if: matrix.os == 'windows-2022'
+        shell: cmd
+        run: |
+          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\vsdevcmd.bat" -arch=amd64 -host_arch=amd64
```
Comment (BillyONeal) [inline .github/workflows/build.yaml:55]:
```suggestion
          call "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\Tools\VsDevCmd.bat" -arch=amd64 -host_arch=amd64
```

capitalization evidence

**Review by BillyONeal — COMMENTED** [review]
Before we can remove the other pipeline run I think we need to see the following to make sure the world works:

* build failure
* <s>formatting failure, and ability to download the resulting diff</s> https://github.com/ras0219-msft/vcpkg-tool/actions/runs/4824887104
* vcpkg-test failure
* vcpkg-artifacts test failure
* e2e test failure

**Comment by BillyONeal** [comment]
Wait why didn't Actions run on this?

File: .github/workflows/build.yaml (hunk 1 above)
Comment (ras0219-msft) [reply .github/workflows/build.yaml:84]: I do think that would be good, but I wanted to do that as a separate PR after pipelines.yml is removed (so we don't have to make the change in two places)

**Review by BillyONeal — APPROVED** [review]
❤️

**Pull request merged**